pkg/gadgets/trace/dns/tracer/tracer.go

```diff
@@ -286,8 +297,11 @@ func (t *Tracer) install() error {
 
 	parseAndEnrichDNSEvent := func(rawSample []byte, netns uint64) (*types.Event, error) {
 		bpfEvent := (*dnsEventT)(unsafe.Pointer(&rawSample[0]))
-		if len(rawSample) < int(unsafe.Sizeof(*bpfEvent)) {
-			return nil, errors.New("invalid sample size")
+		// TODO: Why do I need 4+?
```

Comment:
I'll have a quick look at this, and then approve the PR. Or did you already understand why?

Reply:
I still don't know. I think there is a header that is added to the sample, however I haven't gone into the details. It's working on main because the check uses `<` instead of `==`, if I switch to the later then it fails as well.

https://github.com/inspektor-gadget/inspektor-gadget/blob/3b33ab1cc7234994bff2726483e735965bb5ae8b/pkg/gadgets/trace/dns/tracer/tracer.go#L289

Reply:
> Mmmm I didn't see why. Maybe padding?

That was my first idea but not, both the golang and c structures have the same memory layout and size.